rust-src/concordium_base/src/protocol_level_tokens/token_transaction.rs

```diff
@@ -0,0 +1,207 @@
+use crate::{
+    common::{
+        cbor,
+        cbor::{
+            CborDecoder, CborDeserialize, CborEncoder, CborSerializationError,
+            CborSerializationResult, CborSerialize, DataItemType,
+        },
+    },
+    protocol_level_tokens::{token_holder::TokenHolder, RawCbor, TokenAmount, TokenId},
+    transactions::Memo,
+};
+use concordium_base_derive::{CborDeserialize, CborSerialize};
+
+/// Embedded CBOR, see <https://www.iana.org/assignments/cbor-tags/cbor-tags.xhtml>
+const CBOR_TAG: u64 = 24;
+
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+/// Payload for token transaction. The transaction is a list of token
+/// operations. Operations includes governance operations, transfers etc.
+pub struct TokenOperationsPayload {
+    /// Id of the token
+    pub token_id:   TokenId,
+    /// Token operations in the transaction
+    pub operations: RawCbor,
+}
+
+impl TokenOperationsPayload {
+    pub fn deserialize_operations(&self) -> CborSerializationResult<TokenOperations> {
```

Comment:
Since this is user facing it would be good to have doc comments and some context to why this is a separate step.